File: frontend/src/interfaces/product.ts
import { Category } from '@/interfaces/category';
import { Tag } from '@/interfaces/tag';

export interface ProductAverageRatings {
  boxRating: number;
  boxReusabilityRating: number;
  productReusabilityRating: number;
}

export interface Product {
  id?: number;
  name: string;
  category: Category;
  tags: Tag[];
  averageRatings: ProductAverageRatings;
}

File: mobile/android/app/src/main/java/com/hackyeah/nowaste/data/model/Product.kt
package com.hackyeah.nowaste.data.model

import java.io.Serializable

class Product(
    val id: Long,
    val barcode: String,
    val name: String,
    val photoUrl: String?,
    val averageRatings: Rating,
    val category: Category,
    val tags: List<Tag>
) : Serializable
